app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/securities/ConfirmFullRepaymentController.scala

```diff
@@ -16,18 +16,142 @@
+  // @uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.securities.ConfirmFullRepaymentController.show(id: String)
+  // @(form: Form[YesNo], securityId: String, totalValue: BigDecimal, postAction: Call)
+  //    (implicit request: RequestWithSessionData[_], messages: Messages, viewConfig: ViewConfig)
+  def show(id: String): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    showForId(request, journey, id)
+  }
+
+  private def showForId(implicit request: Request[_], journey: SecuritiesJourney, id: String) =
+    journey
+      .getDisplayDeclarationIfValidSecurityDepositId(id)
+      .flatMap(_.getSecurityDetailsFor(id).map(_.amountPaid))
+      .flatMap(x => Try(BigDecimal(x)).toOption)
+      .fold(errorHandler.errorResult()) { amountPaid =>
+        Ok(
+          confirmFullRepaymentPage(
+            form.withDefault(
+              journey.answers.reclaimingFullAmount
+                .fold(Option.empty[YesNo])(_ => Some(YesNo.Yes))
+            ),
+            id,
+            amountPaid,
+            routes.ConfirmFullRepaymentController.submit(id)
+          )
+        )
+      }
+      .asFuture
+
+  // POST         /securities/confirm-full-repayment/:id
+  // @uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.securities.ConfirmFullRepaymentController.submit(id: String)
+  def submit(id: String): Action[AnyContent] = actionReadWriteJourney { implicit request => journey =>
+    form.bindFromRequest
+      .fold(
+        formWithErrors =>
+          (
+            journey,
+            journey
+              .getDisplayDeclarationIfValidSecurityDepositId(id)
+              .flatMap(_.getSecurityDetailsFor(id).map(_.amountPaid))
+              .flatMap(x => Try(BigDecimal(x)).toOption)
+              .map(amountPaid =>
+                BadRequest(
+                  confirmFullRepaymentPage(
+                    formWithErrors,
+                    id,
+                    amountPaid,
+                    routes.ConfirmFullRepaymentController.submit(id)
+                  )
+                )
+              )
+              .getOrElse(errorHandler.errorResult())
+          ).asFuture,
+        {
+          case Yes =>
+            submitYes(id, journey)
+          case No  =>
+            submitNo(id, journey)
+        }
+      )
+  }
+
+  def submitYes(securityId: String, journey: SecuritiesJourney)(implicit
+    request: Request[_]
+  ): Future[(SecuritiesJourney, Result)] =
+    journey
+      .submitFullAmountsForReclaim(securityId)
+      .fold(
+        { error =>
+          logger.warn(error)
+          (journey, errorHandler.errorResult())
+        },
+        updatedJourney => (updatedJourney, Redirect(routes.ChooseFileTypeController.submit()))
+      )
+      .asFuture
+
+  def submitNo(securityId: String, journey: SecuritiesJourney)(implicit
+    request: Request[_]
+  ): Future[(SecuritiesJourney, Result)] =
+    journey
+      .clearReclaimAmount(securityId)
```

Comment:
this violates AC7: When changing answer from 'Yes' to 'No' pre-populate checkboxes and input boxes in /select-duties & /enter-claim pages

Reply:
we should keep full amount claims as it is

Reply:
in that case `isFullSecurityAmountClaimed` will return true, will this cause any issues? will we skip pages?

Reply:
we have to design the journey to deal with this, probably we need another special flag, like `changeClaimDetailsMode
`

Reply:
Understood, I will remove this

Reply:
this has been removed in the latest commit